http_proxy_errors.go

```diff
@@ -146,7 +146,7 @@ func handleNetError(req *http.Request, err error) (code int, msg, label string)
 }
 
 func handleTLSRecordHeader(req *http.Request, err error) (code int, msg, label string) {
-	var headerErr *tls.RecordHeaderError
+	var headerErr tls.RecordHeaderError
```

Comment:
Looks like we could have some testing around it.

Reply:
Added tests for record header and unverified cert.